packages/dds/tree/src/core/tree/detachedFieldIndex.ts

```diff
@@ -35,15 +38,43 @@ import { IIdCompressor } from "@fluidframework/id-compressor";
  */
 export type ForestRootId = Brand<number, "tree.ForestRootId">;
 
+/**
+ * fake revision used to mark that the revision stored in a {@link DetachedFieldIndex} is not yet
+ * set after loading data from a summary
+ */
+export const fakeRevisionWhenNotSet = 33 as SessionSpaceCompressedId;
```

Comment:
I'm worried 33 with collide with an actual number. It's also not very clear, when encountering 33 in a debug scenario, that 33 is a special value. Let's use `Number.NaN` for now. You'll need to update the revision comparison code to use `Object.is()` instead of `===` in `setRevisionsForLoadedData`.

Longer term, I hope we can get rid of the need for this, or transition to using `undefined`.